File: algorithm/链表常见题/链表元素删除/removeElements.ts
import ListNode from "dataStructure/链表/ListNode"

function removeElements2(head: ListNode, val: number): ListNode {
    let dummyHead: ListNode = new ListNode(0)
    dummyHead.next = head
    let pre: ListNode = dummyHead
    let cur: ListNode = dummyHead.next
    while (cur != null) {
      if (cur.val == val) {
        pre.next = cur.next
      }
      cur = cur.next!
      pre = pre.next!
    }
    return dummyHead.next
}

let linkedlist = new ListNode(1, 
  new ListNode(2, 
    new ListNode(6, 
      new ListNode(3, 
        new ListNode(4, 
          new ListNode(5, 
            new ListNode(6)))))))



removeElements2(linkedlist, 6)

console.log("检查一下ts是不是运行了")

File: algorithm/链表常见题/链表元素删除/removeElements.java
package algorithm.链表常见题.链表元素删除;

import dataStructure.链表.ListNode;

/**
 * 
 * [203. 移除链表元素](https://leetcode.cn/problems/remove-linked-list-elements/submissions/623091945/)
 * 
 * 解法一：我自己写的，利用2个指针，前pre，当前cur，当前的值等于目标值，移动pre的next指针，否则移动pre指针
 * 
 */

public class removeElements {
  

  public ListNode removeElements(ListNode head, int val) {
    ListNode dummyHead = new ListNode(0);
    dummyHead.next = head;
    ListNode pre = dummyHead;
    ListNode cur = dummyHead.next;
    while (cur != null) {
      if (cur.val == val) {
        pre.next = cur.next;
      } else {
        pre = pre.next;
      }
      cur = cur.next;
    }
    return dummyHead.next;
  }
}
